### PR TITLE
New programs

### PR DESCRIPTION
**Pull request by kalbasit**
This is a first of a series of batches that I'll submit as I clean up my own soxin directory.

https://github.com/kalbasit/soxincfg/pull/17 updates my cfg to work with this PR.

### CODE REVIEW

File: modules/programs/git/default.nix [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+    (optionalAttrs (mode == "home-manager") {
+      programs.git = {
+        inherit (cfg) package userName userEmail;
+
+        enable = true;
+
+        signing = mkIf (cfg.gpgSigningKey != null) {
+          key = cfg.gpgSigningKey;
+          signByDefault = true;
```
Comment (kalbasit) [inline modules/programs/git/default.nix:55]: should this be exposed?

File: modules/programs/autorandr/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+with lib;
+let
+  cfg = config.soxin.programs.autorandr;
+in
+{
+  options = {
+    soxin.programs.autorandr = {
+      enable = mkEnableOption ''
+        enable handling of hotplug and sleep events by autorandr.</para><para>
```
Comment (rissson) [inline modules/programs/autorandr/default.nix:11]: This may not render properly.

File: modules/programs/git/default.nix (hunk 1 above)
Comment (rissson) [reply modules/programs/git/default.nix:55]: I don't think so. Maybe we should `mkDefault` it though.

File: modules/programs/keybase/default.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{ mode, config, pkgs, lib, ... }:
+
+with lib;
+let
+  cfg = config.soxin.programs.keybase;
+in
+{
+  options = {
+    soxin.programs.keybase = {
+      enable = mkEnableOption "enable Keybase.";
```
Comment (rissson) [inline modules/programs/keybase/default.nix:10]:
```suggestion
      enable = mkEnableOption "Keybase";
```

File: modules/programs/keybase/default.nix [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+      enableFs =
+        recursiveUpdate
+          (mkEnableOption "enable Keybase filesystem")
```
Comment (rissson) [inline modules/programs/keybase/default.nix:13]:
```suggestion
          (mkEnableOption "Keybase filesystem")
```

File: modules/programs/less/default.nix [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    s forw-line
+    r back-line
+    n right-scroll
+    ' repeat-search
+    ? reverse-search
+  '';
+
+in
+{
+  options.soxin.programs.less = {
```
Comment (rissson) [inline modules/programs/less/default.nix:33]: Don't we want to use `mkSoxinModule` to be able to override the keyboard layout here?

File: modules/programs/starship/default.nix [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        if [ -z "$INSIDE_EMACS" ]; then
+          eval "$(${pkgs.starship}/bin/starship init zsh)"
+        fi
+      '';
+    })
+
+    (optionalAttrs (mode == "home-manager") {
+      programs.starship = {
+        enable = true;
+        enableZshIntegration = true;
```
Comment (rissson) [inline modules/programs/starship/default.nix:26]:
```suggestion
        enableZshIntegration = config.soxin.programs.zsh.enable;
```

**Review by rissson — CHANGES_REQUESTED** [review]

File: modules/programs/git/default.nix (hunk 1 above)
Comment (kalbasit) [reply modules/programs/git/default.nix:55]: done

File: modules/programs/autorandr/default.nix (hunk 2 above)
Comment (kalbasit) [reply modules/programs/autorandr/default.nix:11]: done

File: modules/programs/less/default.nix (hunk 5 above)
Comment (kalbasit) [reply modules/programs/less/default.nix]: done

**Review by rissson — APPROVED** [review]

**Review by shahrukh330 — APPROVED** [review]

**Pull request merged**